src/js/views/postTweet.js

```diff
@@ -1,48 +1,62 @@
 import React from "react";
 import TwitterService from '../services/twitterService.js';
+import PubSubListener from './pubSubListener.js';
 
 class PostTweetUI extends React.Component {
   constructor(props){
     super(props);
     this.state = {
       numChars: 0,
       postTweetButtonDisabled: true,
-      userTweetText: '',
+      postTweetText: props.postTweetText,
       postStatus: 'pending',
     }
   }
 
+  componentWillReceiveProps(nextProps){
+    if(nextProps.postTweetText !== this.state.postTweetText) {
+      this.setState({
+        numChars: 0,
+        postTweetButtonDisabled: true,
```

Comment:
you shouldn't need postTweetButtonDisabled to be explicit, you can just base it on the counter or postTweetText